cmd/appgw-ingress/main.go

```diff
@@ -178,13 +177,26 @@ func getAzAuth(vars environment.EnvVariables) (autorest.Authorizer, error) {
 
 func waitForAzureAuth(envVars environment.EnvVariables, client n.ApplicationGatewaysClient) {
 	const retryTime = 10 * time.Second
-	for counter := 0; counter <= maxAuthRetry; counter++ {
-		if _, err := client.Get(context.Background(), envVars.ResourceGroupName, envVars.AppGwName); err != nil {
-			glog.Error("Error getting Application Gateway", envVars.AppGwName, err)
-			glog.Infof("Retrying in %v", retryTime)
-			time.Sleep(retryTime)
+	for {
```

Comment:
I would like to keep the `maxAuthRetry` thing here. Unless is some sort of an event loop, loops should not be infinite :)

We should bring `maxAuthRetry` back **AND** we should make it so once we reach that count we then print the nice log statement you have composed.
I can imagine - in the current state of this function - even if there's a delay of propagating credentials etc - we would print this statement. It'd be nice to try a few times (or `maxAuthRetry` times) before advising the customer what to do.  In cases where we eventually get a 2xx from this - we would have logged a bunch of alarming statements.

Here is why I believe we should eventually terminate this loop and crash:
  - keeping the container/pod healthy while we keep getting 403 and logging this is providing a false sense of normalcy - `kubectl get pods` would show `status=Running` and it would appear as if all is good. Crashing the container eventually, would most likely result in `kubectl logs …`, which how the customer would get to see this message below.

  - it seems that this log message would be emitted infinitely every 10 seconds, until the 403 is resolved or the pod deleted. That could result in something like 3 MB of logs per day per AGIC, which seems excessive.

Reply:
Good point!